src/core/network.js

```diff
@@ -414,9 +415,27 @@ PDFNetworkStreamFullRequestReader.prototype = {
       networkManager.abortRequest(fullRequestXhrId);
     }
 
+    // Content-Disposition: attachment; filename=Naïve file.txt
+    var fullRequestXhr = networkManager.getRequestXhr(fullRequestXhrId);
+    var contentDisposition =
+      fullRequestXhr.getResponseHeader('Content-Disposition');
+    this._fileName = this._extractNameFrom(contentDisposition);
+
     this._headersReceivedCapability.resolve();
   },
 
+  _extractNameFrom:
+      function PDFNetworkStreamFullRequestReader_extractNameFrom(cDisposition) {
+    if (cDisposition) {
+      var parts = /.+; filename="(.+)"/gi.exec(cDisposition);
```

Comment:
I'm afraid that you might be missing the point here, since as-is this patch would allow a completely bogus filename to be used in various places the viewer; one example being when downloading a file.

Note how in [web/app.js](https://github.com/mozilla/pdf.js/blob/master/web/app.js), we're using the [`getPDFFileNameFromURL`](https://github.com/mozilla/pdf.js/blob/master/web/ui_utils.js#L394) function to obtain a `*.pdf` filename (with a fallback to `document.pdf`).
So I definitely think that some form of validation of a Content-Disposition filename is necessary, if not on the API side, then at the *very* least not using it unconditionally in the viewer (like the patch does)!
Without validation on the API side, then [web/app.js#L731](https://github.com/mozilla/pdf.js/blob/master/web/app.js#L731) would probably need to look something like this:
```javascript
let filename;
if (this.attachedFileName) {
  filename = getPDFFileNameFromURL(this.attachedFileName, '');
}
filename = filename || getPDFFileNameFromURL(this.url);
```